app/views/contexts/preview/preview_question.py

```diff
@@ -0,0 +1,65 @@
+from typing import Any, Union
+
+from werkzeug.datastructures import ImmutableDict
+
+
+class PreviewQuestion:
+    def __init__(
+        self,
+        *,
+        block: ImmutableDict,
+    ):
+        self._block = block
+        self._block_id = block["id"]
+        self._question = self.get_question()
+        self._title = self._question["title"]
+        self._answers = self._build_answers()
+        self._descriptions = self._question.get("description")
+        self._guidance = self._question.get("guidance")
+        self._instruction = self._question.get("instruction")
+        self._type = self._question.get("type")
+
+    def _build_answers(self) -> list[dict]:
+        answers_list = []
+        for answer in self._question.get("answers", []):
+            answer_dict = {}
+            if answer_label := answer.get("label"):
+                answer_dict["label"] = answer_label
+
+            if options := answer.get("options"):
+                options_list = [option["label"] for option in options]
+                answer_dict["options"] = options_list
+
+            if description := answer.get("description"):
+                answer_dict["description"] = description
+
+            if instruction := answer.get("instruction"):
+                answer_dict["instruction"] = instruction
+
+            if guidance := answer.get("guidance"):
+                answer_dict["guidance"] = guidance
```

Comment:
Minor, can be simplified:
```suggestion

            keys = ["label", "description", "instruction", "guidance"]
            for key in keys:
                if value := answer.get(key):
                    answer_dict[key] = value

            if options := answer.get("options"):
                options_list = [option["label"] for option in options]
                answer_dict["options"] = options_list
```